ci-operator/step-registry/aws/provision/bastionhost/aws-provision-bastionhost-commands.sh

```diff
@@ -34,15 +34,20 @@ if [[ ! -f "${SHARED_DIR}/vpc_id" && ! -f "${SHARED_DIR}/public_subnet_ids" ]];
   # for OCP
   echo "Reading infra id from file metadata.json"
   infra_id=$(jq -r '.infraID' ${SHARED_DIR}/metadata.json)
-  echo "Looking up IDs for VPC ${infra_id} and subnet ${infra_id}-public-${REGION}a"
-  VpcId=$(aws --region ${REGION} ec2 describe-vpcs --filters Name=tag:"Name",Values=${infra_id}-vpc --query 'Vpcs[0].VpcId' --output text)
-  PublicSubnet=$(aws --region ${REGION} ec2 describe-subnets --filters "Name=tag:kubernetes.io/cluster/${infra_id},Values=owned" "Name=tag:Name,Values=*public*" --query 'Subnets[0].SubnetId' --output text)
+  vpc_name="${infra_id}-vpc"
+  public_subnet_name="${infra_id}-subnet-public-${REGION}a"
+  echo "Looking up IDs for VPC ${vpc_name} and subnet ${public_subnet_name}"
+  VpcId=$(aws --region ${REGION} ec2 describe-vpcs --filters Name=tag:"Name",Values=${vpc_name} --query 'Vpcs[0].VpcId' --output text)
+  PublicSubnet=$(aws --region ${REGION} ec2 describe-subnets --filters "Name=tag:kubernetes.io/cluster/${infra_id},Values=owned" "Name=tag:Name,Values=${public_subnet_name}" --query 'Subnets[0].SubnetId' --output text)
```

Comment:
This part code doens't work for 4.16 job, we can see errors from bastionhost log in https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift_release/55740/rehearse-55740-periodic-ci-openshift-release-master-nightly-4.16-e2e-aws-ovn-ha-cert-rotation-suspend-30d/1828832885851820032/artifacts/e2e-aws-ovn-ha-cert-rotation-suspend-30d/aws-provision-bastionhost/build-log.txt

Reply:
That's b/c zones were specified in [install-config](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift_release/55740/rehearse-55740-periodic-ci-openshift-release-master-nightly-4.16-e2e-aws-ovn-ha-cert-rotation-suspend-30d/1828832885851820032/artifacts/e2e-aws-ovn-ha-cert-rotation-suspend-30d/ipi-install-install/build-log.txt), no `${REGION}a`:
```
controlPlane:
  platform:
    aws:
      zones:
      - us-west-2c
      - us-west-2d
      type: m6a.xlarge
  architecture: amd64
  name: master
  replicas: 3
compute:
- platform:
    aws:
      zones:
      - us-west-2c
      - us-west-2d
      type: m6a.xlarge
  architecture: amd64
  name: worker
  replicas: 3
```

Reply:
Since 4.16, the default subnets name has changed:
4.16 and above: `${infra_id}-subnet-public-${REGION}[abc...]`
4.15 and below: `${infra_id}-public-${REGION}[abc...]`

So I think the old filter is a good way: `--filters "Name=tag:kubernetes.io/cluster/${infra_id},Values=owned" "Name=tag:Name,Values=*public*"`

Reply:
oooh, right, I'll add a comment to stop touching this part :)